scripts/proteinInteractionEBI/parseEBI.py

```diff
@@ -0,0 +1,331 @@
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        # idString example: "MI:0000"
+        idString = termText.split('\n')[1].split(' ')[1]
+        className = get_class_name(termText.split('\n')[2].split(': ')[1])
+        id2className[idString] = className
+        id2node[idString] = Node(idString)
+
+    # build the parent-child relation at the second iteration
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        termList = termText.split('\n')
+        idString = termList[1].split(' ')[1]
+        parentIdList = get_parent_id_list(termList[1:])
+        for pId in parentIdList:
+            id2node[pId].childList.append(id2node[idString])
+            id2node[idString].parentList.append(id2node[pId])
+
+    # get the idStrings for the three target set
+    dfsCaller = GetTreeValues()
+    interactionTypeIdSet = dfsCaller.get_subset_id(id2node['MI:0001']) # root id: MI:0001 
+    detectionMethodIdSet = dfsCaller.get_subset_id(id2node['MI:0190'])# root id: MI:0190
+    interactionSourceIdset = dfsCaller.get_subset_id(id2node['MI:0444']) # root id: MI:0444
+
+    # delete root node value from the set
+    interactionTypeIdSet.remove('MI:0001')
+    detectionMethodIdSet.remove('MI:0190')
+    interactionSourceIdset.remove('MI:0444')
+
+    setList = [interactionTypeIdSet, detectionMethodIdSet,interactionSourceIdset]
+    print('The schema amount of each Enum: interactionType, detectionMethod,interactionSource are ')
+    print([len(s) for s in setList])
+
+    schemaList = []
+    psimi2dcid = []
+    newSourceMap = {'references':{}}
+    for termText in fileTerms:
+        if not termText.startswith('[Term]'):
+            continue
+        term = termText.split('\n')[1:]
+        schemaRes = get_schema_from_text(term,id2node,newSourceMap,id2className,
+                                      interactionTypeIdSet,detectionMethodIdSet,interactionSourceIdset)
+        if schemaRes:
+            schema, psimi, dcid, newSourceMap  = schemaRes
+            schemaList.append(schema)
+            psimi2dcid.append(psimi+': ' + dcid)
+    schemaEnumText = '\n\n'.join(schemaList)
+    schema = '# This schema file is generated by parseEBI.py. Please don\'t edit.\n'
+    schemaEnumText = schema + schemaEnumText
+
+    with open('BioOntologySchemaEnum.mcf','w') as fp:
+        fp.write(schemaEnumText)
+    with open('psimi2dcid.txt','w') as fp:
+        fp.write('\n'.join(psimi2dcid))
+        
+    writeList = []
+    for sourceType in newSourceMap:
+        if not newSourceMap[sourceType]: continue
+        writeList.append(sourceType)
+        for source in newSourceMap[sourceType]:
+            line = source + ': ' + newSourceMap[sourceType][source]
+            writeList.append(sourceType)
+        writeList.append('\n')
+    if writeList:
+        with open('BioEBINewSource.txt','w') as fp:
+            fp.write('\n'.join(writeList))
+
+
```

Comment:
Try to use absl library so you get FLAGS. That's a much nicer way to take arguments.

Reply:
Here is an example: https://github.com/datacommonsorg/data/blob/master/scripts/us_fed/treasury_constant_maturity_rates/generate_csv_and_mcf.py#L31-L41